meta-xt-control-domain/recipes-guest/doma/u-boot-android-bazel_1.0.0.bb

```diff
@@ -0,0 +1,75 @@
+SUMMARY = "Deployment of Android U-Boot, which is built by the Bazel build system."
+DESCRIPTION = "Android U-Boot is a standard tool to boot Android."
+
+inherit base
+
+# PARAMETERS
+
+UBOOT_REPO_GIT_URL ?= "github.com/xen-troops/android_u-boot_manifest"
+
+# TBD. Find a way to form this variable automatically from the
+# "UBOOT_REPO_GIT_URL" parameter
+UBOOT_REPO_GIT_URL_DOTTED ?= "github.com.xen-troops.android_u-boot_manifest"
+
+UBOOT_REPO_DOWNLOAD_PROTOCOL ?= "https"
+UBOOT_REPO_GIT_BRANCH ?= "xenvm-trout-main"
+UBOOT_REPO_MANIFEST ?= "default.xml"
+U_BOOT_BUILD_TARGET ?= "xen-guest-android-virtio_aarch64"
+
+# IMPLEMENTATION
+
+UBOOT_DOWNLOAD_REPO_DIR = "${DL_DIR}/repo/${UBOOT_REPO_GIT_URL_DOTTED}/${UBOOT_REPO_MANIFEST}"
+UBOOT_LOCAL_REPO_RELATIVE_PATH = "repo/${UBOOT_REPO_GIT_URL_DOTTED}/${UBOOT_REPO_MANIFEST}"
+UBOOT_LOCAL_REPO_DIR = "${WORKDIR}/${UBOOT_LOCAL_REPO_RELATIVE_PATH}"
+UBOOT_BUILD_DIR = "${B}/${UBOOT_REPO_GIT_URL_DOTTED}/${UBOOT_REPO_MANIFEST}/${U_BOOT_BUILD_TARGET}"
+
+SRC_URI = "\
+           repo://${UBOOT_REPO_GIT_URL};protocol=${UBOOT_REPO_DOWNLOAD_PROTOCOL};branch=${UBOOT_REPO_GIT_BRANCH};manifest=${UBOOT_REPO_MANIFEST}  \
+           "
+
+LICENSE = "GPLv2+"
+LIC_FILES_CHKSUM = "file://${WORKDIR}/${UBOOT_LOCAL_REPO_RELATIVE_PATH}/repo/u-boot/Licenses/README;md5=2ca5f2c35c8cc335f0a19756634782f1"
+
+FILES:${PN} = "\
+               ${libdir}/xen/boot/u-boot-doma \
+               "
+
+do_unpack() {
+    if ! [ -d ${UBOOT_LOCAL_REPO_DIR} ]; then
+        mkdir -p ${UBOOT_LOCAL_REPO_DIR};
+        cp -r ${UBOOT_DOWNLOAD_REPO_DIR}/* ${UBOOT_LOCAL_REPO_DIR}/;
+    fi
+}
+
+do_compile() {
+    cd ${UBOOT_LOCAL_REPO_DIR}/repo;
+    export CC=""
+    export CXX=""
+    export LD=""
+    export LDFLAGS=""
+    export CFLAGS=""
+    export CXXFLAGS=""
+    # Use it instead if you want to get more debug build artifacts.
+    # tools/bazel run //u-boot:${U_BOOT_BUILD_TARGET}_dist --verbose_failures --sandbox_debug -- --dist_dir=${UBOOT_BUILD_DIR};
+    tools/bazel run //u-boot:${U_BOOT_BUILD_TARGET}_dist -- --dist_dir=${UBOOT_BUILD_DIR};
+    tools/bazel shutdown;
```

Comment:
AFAIK, bazel supports auto shutdown

Reply:
Hi @lorc,

I've looked at the moulin Bazel moulin builder:
https://github.com/xen-troops/moulin/pull/77/files#diff-5de43ce242b0cf99cc9dfb32b6be11b9b43501fe68ed6ee32a98b9a11ead6dc7R74

It is using a separate Bazel shutdown call. So, I was not able to find some separate flag, which would auto shutdown the Bazel server after the build is over. I've refactored the code to do the same thing as mentioned builder. That allows us to avoid prepending do_clean, but shutdown call is still used.

Done.